Userland/Libraries/LibC/sys/select.cpp

```diff
@@ -25,8 +28,53 @@ int select(int nfds, fd_set* readfds, fd_set* writefds, fd_set* exceptfds, timev
 
 int pselect(int nfds, fd_set* readfds, fd_set* writefds, fd_set* exceptfds, const timespec* timeout, const sigset_t* sigmask)
 {
-    Syscall::SC_select_params params { nfds, readfds, writefds, exceptfds, timeout, sigmask };
-    int rc = syscall(SC_select, &params);
-    __RETURN_WITH_ERRNO(rc, rc, -1);
+    if (nfds < 0 || nfds >= FD_SETSIZE)
+        return EINVAL;
+
+    Vector<pollfd, FD_SETSIZE> fds;
+    for (int fd = 0; fd < nfds; fd++) {
+        short events = 0;
+        if (readfds && FD_ISSET(fd, readfds))
+            events |= POLLIN;
+        if (writefds && FD_ISSET(fd, writefds))
+            events |= POLLOUT;
+        if (exceptfds && FD_ISSET(fd, exceptfds))
+            events |= POLLPRI;
+        if (!events)
+            continue;
+
+        fds.append({ fd, events, 0 });
```

Comment:
Can this append fail?
If yes:
* use try append and propagate ENOMEM
If no:
* use unsafe_append

But seems like the NO case

Reply:
I'm allocating the array inline with `Vector<pollfd, FD_SETSIZE>`. It shouldn't fail since the wrapper rejects sets larger than `FD_SETSIZE`, but I'll substitute `append()` with `unchecked_append()`.

Worst case is a userland program blows up because they used `select()`, so it's not _that_ much of a problem given the context of this PR.